### ISSUE TITLE
Code separate!

### ISSUE THREAD

**Issue by janryWang** [dtao/lazy.js#188]
I need Lazy.js like lodash.js as modular  separate code, rather than all at once import, in fact, I am more concerned about its core modules Squence

**Comment by xgbuils** [comment]
Hi! Maybe you are interested in [iterum](https://github.com/xgbuils/iterum). It is es6 modular library that uses iterables to achieve lazy evaluation.

**Comment by odin3** [comment]
@xgbuils, is it stable enough?

**Comment by xgbuils** [comment]
Yes. i'm working on the new major release that implements new combinatorial functions and brings support for infinite iterables in all implemented methods.

**Comment by odin3** [comment]
@xgbuils, and what about performance? TypeScript support?

**Comment by xgbuils** [comment]
You can see that exploring [iterum dev-2.0.0 branch](https://github.com/xgbuils/iterum/tree/dev-2.0.0/)

About performance, I'm trying to improve that. Using native generators for lazy evaluation has some caveats that another library [imlazy](https://github.com/benji6/imlazy) is trying to solve and I will try to implement as soon as possible.

About TypeScript, never I worked with it and I don't know if it has support.

There are other libraries that you can explore:

- [ramda](https://github.com/ramda/ramda/)
- [imlazy](https://github.com/benji6/imlazy)

Don't hesitate to open issues and contribute.

Cheers!

**Comment by odin3** [comment]
@xgbuils, thank you!
